packages/framework-core/src/services/graphql/graphql-generator.ts

```diff
@@ -198,13 +221,78 @@ export class GraphQLGenerator {
       sortBy: {},
     }
   }
+
+  private static getFields(info: GraphQLResolveInfo): string[] {
+    let fields: string[] = []
+
+    const isList = (type: any): boolean => {
+      if (type instanceof GraphQLNonNull) {
+        return type.ofType instanceof GraphQLList
+      }
+      return type instanceof GraphQLList
+    }
+
+    const extractFields = (
+      selectionSet: SelectionSetNode,
+      path: string[] = [],
+      parentType: GraphQLObjectType | any
+    ): string[] => {
+      let subFields: string[] = []
+      if (selectionSet && selectionSet.selections) {
+        selectionSet.selections.forEach((selection: any) => {
+          const fieldName = selection.name.value
+          const field = parentType.getFields()[fieldName]
+
+          if (!field) {
+            console.error(`Field ${fieldName} not found on type ${parentType.name}. Skipping.`) // @TODO: remove
+            return
+          }
+
+          const fieldType: GraphQLNamedInputType = getNamedType(field.type as GraphQLNamedInputType)
+          const currentPath = [...path, fieldName]
+
+          if (isList(field.type)) {
+            const elementType = getNamedType(field.type.ofType)
+            if (isObjectType(elementType)) {
+              currentPath[currentPath.length - 1] += '[]'
+            }
+          }
+
+          if (!selection.selectionSet) {
+            subFields.push(currentPath.join('.'))
+          } else {
+            const nextParentType = isObjectType(fieldType) ? fieldType : parentType
+            subFields = subFields.concat(extractFields(selection.selectionSet, currentPath, nextParentType))
+          }
+        })
+      }
+      return subFields
+    }
+
+    info.fieldNodes.forEach((fieldNode: FieldNode) => {
+      const firstField: string = fieldNode.name.value
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      const firstFieldType = info.schema.getQueryType().getFields()[firstField].type
+      const rootType: GraphQLObjectType = getNamedType(
+        firstFieldType as GraphQLInputType
+      ) as unknown as GraphQLObjectType
+
+      if (fieldNode.selectionSet) {
+        fields = fields.concat(extractFields(fieldNode.selectionSet, [], rootType))
+      }
+    })
+
+    return fields
+  }
 }
```

Comment:
Amazing. I though it was going to be much easier to get the selection set from the graphQL query, but you needed to implement this crazy recursive method. Great job!